src/test/java/org/opensearch/security/test/helper/rest/RestHelper.java

```diff
@@ -296,40 +374,54 @@ protected final CloseableHttpClient getHTTPClient() throws Exception {
 				protocols = new String[] { "TLSv1", "TLSv1.1", "TLSv1.2" };
 			}
 
-			final SSLConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(sslContext, protocols, null,
-					NoopHostnameVerifier.INSTANCE);
+			final TlsStrategy tlsStrategy = ClientTlsStrategyBuilder
+					.create()
+					.setSslContext(sslContext)
+					.setTlsVersions(protocols)
+					.setHostnameVerifier(NoopHostnameVerifier.INSTANCE)
+					// See please https://issues.apache.org/jira/browse/HTTPCLIENT-2219
+					.setTlsDetailsFactory(new Factory<SSLEngine, TlsDetails>() {
+						@Override
+						public TlsDetails create(final SSLEngine sslEngine) {
+							return new TlsDetails(sslEngine.getSession(), sslEngine.getApplicationProtocol());
+						}
+					})
+					.build();
 
-			final HttpClientConnectionManager cm = PoolingHttpClientConnectionManagerBuilder.create()
-					.setSSLSocketFactory(sslsf)
-					.setDefaultSocketConfig(SocketConfig.custom().setSoTimeout(60, TimeUnit.SECONDS).build())
+			final AsyncClientConnectionManager cm = PoolingAsyncClientConnectionManagerBuilder.create()
+					.setTlsStrategy(tlsStrategy)
 					.build();
+
 			hcb.setConnectionManager(cm);
 		}
 
-		return hcb.build();
+		final RequestConfig.Builder requestConfigBuilder = RequestConfig.custom()
+				.setResponseTimeout(Timeout.ofSeconds(60));
+
+		return hcb.setDefaultRequestConfig(requestConfigBuilder.build()).disableAutomaticRetries().build();
```

Comment:
Was `disableAutomaticRetries()` added because of issues with any test?

Leaving a link to the Default class in HttpComponents-client5 to come back to later to see the list of default non-retriable and retriable codes: https://hc.apache.org/httpcomponents-client-5.1.x/current/httpclient5/apidocs/org/apache/hc/client5/http/impl/DefaultHttpRequestRetryStrategy.html 

> Create the HTTP request retry strategy with a max retry count of 1, default retry interval of 1 second, and using the following list of non-retriable I/O exception classes:
InterruptedIOException UnknownHostException ConnectException ConnectionClosedException SSLException and retriable HTTP status codes:
SC_TOO_MANY_REQUESTS (429) SC_SERVICE_UNAVAILABLE (503)

Reply:
> Was `disableAutomaticRetries()` added because of issues with any test?
> 
> Leaving a link to the Default class in HttpComponents-client5 to come back to later to see the list of default non-retriable and retriable codes: https://hc.apache.org/httpcomponents-client-5.1.x/current/httpclient5/apidocs/org/apache/hc/client5/http/impl/DefaultHttpRequestRetryStrategy.html
> 
> > Create the HTTP request retry strategy with a max retry count of 1, default retry interval of 1 second, and using the following list of non-retriable I/O exception classes:
> > InterruptedIOException UnknownHostException ConnectException ConnectionClosedException SSLException and retriable HTTP status codes:
> > SC_TOO_MANY_REQUESTS (429) SC_SERVICE_UNAVAILABLE (503)

Not really in the `security` plugin tests, but it is not needed (enabling retries won't harm here)